### DISCUSSION TITLE
How to stop automatic browser opening when error occurs?

### DISCUSSION THREAD

**Discussion by monolithed** [vercel/next.js#50328] [Help]
### Summary

Every time I write code, a browser window with an error opens. Errors usually occur when I pause while writing code or when I reference non-existent variables. How can I disable this option?

### Additional information

It's ok to be notified when error occurs, but please don't touch the system focus

### Example

<img width="1104" alt="Screenshot 2023-05-25 at 19 37 14" src="https://github.com/vercel/next.js/assets/803674/af88268f-ccb3-4872-85f7-14c8dbdbcea9">

macOS: 13.1, Safari 16.2

UPD: The issue occurs only when Safari is open and used as the default browser.

**Comment by icyJoseph** [comment]
Oh wow, this doesn't happen to me - also in MacOS (13.3.1) - could it be your text editor doing this?

**Reply by monolithed** [reply]
@icyJoseph, I'm using Fleet, but before creating the issue, I tried to save a file in vim, and I got the same result - the focus jumped to the browser.

**Reply by monolithed** [reply]
The issue occurs only when Safari is open and used as the default browser.

**Reply by icyJoseph** [reply]
I assumed you were using a built-in terminal, what terminal do you use?

**Reply by monolithed** [reply]
Yep, I use the built-in terminal

**Reply by icyJoseph** [reply]
Mmm, and even when you used vim it happened you said? This is strange.

**Reply by monolithed** [reply]
You're right, I checked that it reproduces in Vim too. I temporarily switched to Chrome, but I'd like to return to Safari.

**Reply by icyJoseph** [reply]
When you used vim, what terminal did you use to run the app?

**Reply by icyJoseph** [reply]
Wow I was able to reproduce this... incredible 😅 then again I haven't used so much of the Mac on my Mac, but this is really odd... I wonder if there's some debug setting or something in Safari?

**Reply by icyJoseph** [reply]
Could this be the reason? https://forums.macrumors.com/threads/safari-auto-reloading-tabs-steal-focus.2320021/

**Reply by icyJoseph** [reply]
Sorry for spamming, but this is really weird... It has to be that somehow, something in the Developer overlay or an uncaught exception, etc, that causes the focus stealing... I would like to try throwing errors and such, and see if the focus is taken away in those cases too.

**Reply by monolithed** [reply]
Same here. it's really weird, and I haven't run into such problems before.
